packages/notifications/README.md

```diff
@@ -0,0 +1,70 @@
+# @clipboard-health/notifications <!-- omit from toc -->
+
+Send notifications through third-party providers.
+
+## Table of contents <!-- omit from toc -->
+
+- [Install](#install)
+- [Usage](#usage)
+  - [`NotificationsClient`](#notificationsclient)
+- [Local development commands](#local-development-commands)
+
+## Install
+
+```bash
+npm install @clipboard-health/notifications
+```
+
+## Usage
+
+### `NotificationsClient`
+
+<embedex source="packages/notifications/examples/notificationClient.ts">
+
+```ts
+import { NotificationClient, type Span } from "@clipboard-health/notifications";
+import { isSuccess } from "@clipboard-health/util-ts";
+
+import { IdempotentKnock } from "../src/lib/internal/idempotentKnock";
+
+const logger = {
+  info: console.log,
+  warn: console.warn,
+  error: console.error,
+} as const;
+const tracer = {
+  trace: <T>(_name: string, _options: unknown, fun: (span?: Span | undefined) => T): T => fun(),
+};
+const client = new NotificationClient({
+  provider: new IdempotentKnock({ apiKey: "test-api-key", logger }),
```

Comment:
I wonder if exposing this possibility of setting provider (and forcing anyone that uses the library to know about and import the idempotent knock) is a complication that is worth it. 

The implementation of the client screams Knock (workflows and triggers) and I don't think we would ever reasonably reuse that NotificationClient with anything different than Knock. 

Maybe if instead of passing the entire provider we could just pass an api key and then notification client would construct IdempotentKnock itself would be easier and good enough?

Reply:
That's how I had it, but it made testing harder. You actually commented on a prior PR for ways to remove the `as` assertions for mocks and this was the change to remove those.

I can make it optional and only pass it in tests.